sympy/tensor/tests/test_indexed.py

```diff
@@ -222,6 +222,17 @@ def test_IndexedBase_assumptions():
     assert A[i] != Indexed(a, i)
 
 
+def test_IndexedBase_assumptions_inheritance():
+    I = Symbol('I', integer=True)
+    I_inherit = IndexedBase(I)
+    I_explicit = IndexedBase('I', integer=True)
+
+    assert I_inherit.is_integer
+    assert I_explicit.is_integer
+    assert I_inherit.assumptions0 == I_explicit.assumptions0
+    assert I_inherit != I_explicit  # Because the underlying symbol is different.
```

Comment:
With the above change, I think these should be the same -- and we want them to be so since once the Symbol passes the "event horizon" it is irrelevant that it was a Symbol. ;-)

Reply:
I agree with this point, also if the label is supposed to only be a `Symbol`, you can use something like `label = Symbol(label.name, assumptions)` , here `assumptions` that has been filtered for the `IndexedBase`.

Reply:
The latest commit fixes this, at least in one of many possible ways.